tkg/client/update_cluster_metadata.go

```diff
@@ -0,0 +1,265 @@
+const tkgBomConfigMapName = "tkg-bom"
+const tkgMetadataConfigMapName = "tkg-metadata"
+const clusterMetadataNamespaceRoleName = "tkg-metadata-reader"
+
+// CreateOrUpdateClusterMetadata will create or update the tkg-bom ConfigMap and tkg-metadata ConfigMap
+func (c *TkgClient) CreateOrUpdateClusterMetadata(regionalClusterClient, currentClusterClient clusterclient.Client, isRegionalCluster bool, edition string, clusterName string, clusterNamespace string) error {
+	type ConfigmapRef struct {
+		Name string `yaml:"name"`
+	}
+
+	type Bom struct {
+		ConfigmapRef *ConfigmapRef `yaml:"configmapRef"`
+	}
+
+	type Infrastructure struct {
+		Provider string `yaml:"provider"`
+	}
+
+	type Cluster struct {
+		Name                string          `yaml:"name"`
+		Type                string          `yaml:"type"`
+		Plan                string          `yaml:"plan"`
+		KubernetesProvider  string          `yaml:"kubernetesProvider"`
+		TkgVersion          string          `yaml:"tkgVersion"`
+		Edition             string          `yaml:"edition"`
+		Infrastructure      *Infrastructure `yaml:"infrastructure"`
+		IsClusterClassBased bool            `yaml:"isClusterClassBased"`
+	}
+
+	type TkgMetadata struct {
+		Cluster *Cluster `yaml:"cluster"`
+		Bom     *Bom     `yaml:"bom"`
+	}
+
+	tkgBomConfigMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: clusterMetadataNamespace,
+			Name:      tkgBomConfigMapName,
+		},
+		Data: make(map[string]string),
+	}
+	bomConfiguration, err := c.tkgBomClient.GetDefaultTkgBOMConfiguration()
+	if err != nil {
+		return errors.Wrapf(err, "cannot get the default bom configuration")
+	}
+	bomConfigurationByte, err := yaml.Marshal(bomConfiguration)
+	if err != nil {
+		return errors.Wrap(err, "unable to yaml marshal default bom configuration")
+	}
+	tkgBomConfigMap.Data["bom.yaml"] = string(bomConfigurationByte)
+
+	clusterType := "workload"
+	if isRegionalCluster {
+		clusterType = "management"
+	}
+
+	// the default edition is tkg because tce edition is deprecating
+	clusterEdition := "tkg"
+	if edition != "" {
+		clusterEdition = edition
+	}
+
+	isClusterClassBased, err := regionalClusterClient.IsClusterClassBased(clusterName, clusterNamespace)
+	if err != nil {
+		return errors.Wrap(err, "unable to determine cluster type")
+	}
+
+	infraProvider, err := regionalClusterClient.GetRegionalClusterDefaultProviderName(clusterctlv1.InfrastructureProviderType)
+	if err != nil {
+		return errors.Wrap(err, "failed to get cluster provider information")
+	}
+	providerName, _, err := ParseProviderName(infraProvider)
+	if err != nil {
+		return errors.Wrap(err, "failed to parse the provider name")
+	}
+
+	kubernetesProvider := "VMware Tanzu Kubernetes Grid"
+	if providerName == "tkg-service-vsphere" {
+		kubernetesProvider = "VMware Tanzu Kubernetes Grid Service for vSphere"
+	}
+
+	clusterlist := &capi.ClusterList{}
+	err = regionalClusterClient.ListResources(clusterlist, &crtclient.ListOptions{})
```

Comment:
Why don't call get Cluster directly?

Reply:
Thanks for pointing this out! This was some leftovers for some verifications that I forgot to cleanup. I will fix this.